libsolidity/interface/SourceReferenceFormatter.cpp

```diff
@@ -49,6 +49,21 @@ void SourceReferenceFormatter::printSourceLocation(
 	if (startLine == endLine)
 	{
 		string line = scanner.lineAtPosition(_location->start);
+
+		int locationLength = endColumn - startColumn;
+		if (locationLength > 150)
+		{
+			line = line.substr(0, startColumn) + line.substr(startColumn, 15) + "..." + line.substr(endColumn - 15, 15) + line.substr(endColumn, line.length() - endColumn);
```

Comment:
Why don't you use `line = line.substr(0, startColumn + 15) + ...`?

Reply:
Oh and perhaps `" ... "` instead of `"..."` would be better readable.

Reply:
yes you are right :)